src/base_footprint.cpp

```diff
@@ -61,141 +62,123 @@ BaseFootprintBroadcaster::BaseFootprintBroadcaster(const rclcpp::NodeOptions &)
         this,
         _1));
   }
+
+  tfBroadcaster_ = std::make_unique<tf2_ros::TransformBroadcaster>(this);
+
+  timer_ = this->create_wall_timer(33ms, std::bind(&BaseFootprintBroadcaster::timerCallback, this));
```

Comment:
I am not in favor of a walltimer, as the publishing frequency should no be affected by the speed of the simulator.

Reply:
Good point, ``rclcpp::create_timer`` seems to do the trick as explained [here](https://answers.ros.org/question/362478/how-should-i-create-a-timer-in-ros2/?answer=362544#post-id-362544)